bulk-model-sync-lib/src/commonMain/kotlin/org/modelix/model/sync/bulk/ModelImporter.kt

```diff
@@ -121,52 +122,127 @@ class ModelImporter(private val root: INode, private val continueOnError: Boolea
     private fun syncChildren(node: INode, data: NodeData, progressReporter: ProgressReporter) {
         val allRoles = (data.children.map { it.role } + node.allChildren.map { it.roleInParent }).distinct()
         for (role in allRoles) {
-            val expectedNodes = data.children.filter { it.role == role }
-            val existingNodes = node.getChildren(role).toList()
+            syncChildRole(data, role, node, progressReporter)
+        }
+    }
 
-            // optimization that uses the bulk operation .addNewChildren
-            if (existingNodes.isEmpty() && expectedNodes.all { originalIdToExisting[it.originalId()] == null }) {
-                node.addNewChildren(role, -1, expectedNodes.map { it.concept?.let { ConceptReference(it) } }).zip(expectedNodes).forEach { (newChild, expected) ->
+    private fun syncChildRole(
+        data: NodeData,
+        role: String?,
+        node: INode,
```

Comment:
Would this be easier to read of that variables was called `parent` or `host` or something alike?

Reply:
This method does not exist anymore.
But I changed it for `fun syncChildren(existingParent: INode, expectedParent: NodeData, ...)